frontend/src/Utils/constants.ts

```diff
@@ -1 +1,9 @@
-export const activePlugins = [{ name: 'console-demo-plugin' }, { name: 'sources', pathPrefix: '/apps' }];
+export type PluginType = {
+  name: string;
+  pathPrefix?: string;
+};
+
+type GetActivePlugins = (isBeta: boolean, appName: string) => Promise<PluginType[]>;
+
+export const getActivePlugins: GetActivePlugins = async (isBeta, appName) =>
+  (await fetch(`${isBeta ? '/beta' : ''}/apps/${appName}/plugins.json`)).json();
```

Comment:
```suggestion
  (await fetch(`${isBeta ? '/beta' : ''}/apps/${appName}/plugins.json`)).json().catch((e) => {
    // eslint-disable-next-line no-console
    console.error('Failed to fetch plugin data', e);
    return [];
  });
```
I think we should catch here, so the following then just operates normally with a plugin state of zero plugins.

Reply:
Yeah, good point. I'll wrap it with `try/catch`. FYI we should switch to axios and start using notifications if some request fails. Not sure about those fired from `navigation.ts`